src/js/components/AutoScrollControl/AutoScrollControl.tsx

```diff
@@ -247,34 +275,53 @@ class AutoScrollControl extends React.PureComponent<IAutoScrollControlProps, IAu
                   </div>
                 )}
                 renderThumb={({ props }) => (
-                  <div className={`autoScrollControlSliderThumb ${hideSliderClass}`}
+                  <div
+                    className={`autoScrollControlSliderThumb ${hideSliderClass}`}
                     {...props}
                   />
                 )}
               />
               <button
                 className="autoScrollControlIncreaseSpeed"
-                onClick={this.setSpeed('increment')}> + </button>
+                onClick={this.setSpeed('increment')}
+              >
+                {' '}
+                +{' '}
+              </button>
             </div>
           </div>
         </div>
         <button
           onClick={this.toggleAutoScrollState}
-          className="autoScrollControlPlayBtn">
-          {isAutoScrolling ?
-            <Pause onClick={this.handleIconClick} /> :
-            <Play onClick={this.handleIconClick} />}
+          className="autoScrollControlPlayBtn"
+        >
+          {isAutoScrolling ? (
+            <Pause onClick={this.handleIconClick} />
+          ) : (
+            <Play onClick={this.handleIconClick} />
+          )}
         </button>
       </div>
-    )
+    );
   }
 }
 
-const mapStateToProps = ({ isAutoScrolling, isLoadingAng }: any) => ({ isAutoScrolling, isLoadingAng })
+const mapStateToProps = ({ isAutoScrolling, isLoadingAng }: any) => ({
+  isAutoScrolling,
+  isLoadingAng,
+});
 
 const mapDispatchToProps: IReduxDispatchProps = {
-  setAutoScrolling
-}
+  setAutoScrolling,
+};
 
 // TODO: finding fix for proper typing this
-export default connect<{}, {}, IAutoScrollControlProps, IAutoScrollControlState>(mapStateToProps as any, mapDispatchToProps)(AutoScrollControl as any);
+export default connect<
```

Comment:
Also, it's good if we can have a ticket to conver this to functional code.

Reply:
https://github.com/KhalisFoundation/sttm-web/issues/1800